Storage/ThirdParty/SwiftData.swift

```diff
@@ -609,13 +624,27 @@ open class ConcreteSQLiteDBConnection: SQLiteDBConnection {
             return nil
         }
 
-        let status = immediately ? sqlite3_close(db) : sqlite3_close_v2(db)
+        var status = sqlite3_close(db)
 
-        // Note that if we use sqlite3_close_v2, this will still return SQLITE_OK even if
-        // there are outstanding prepared statements
         if status != SQLITE_OK {
             log.error("Got status \(status) while attempting to close.")
-            return createErr("During: closing database with flags", status: Int(status))
+            SentryIntegration.shared.sendWithStacktrace(message: "Got status \(status) while attempting to close.", tag: "SwiftData", severity: .error)
+            
+            if immediately {
+                return createErr("During: closing database with flags", status: Int(status))
+            }
+            
+            // Note that if we use sqlite3_close_v2, this will still return SQLITE_OK even if
+            // there are outstanding prepared statements
+            status = sqlite3_close_v2(db)
+            
+            if status != SQLITE_OK {
+                
+                // Based on the above comment regarding sqlite3_close_v2, this shouldn't happen.
+                log.error("Got status \(status) while attempting to close_v2.")
+                SentryIntegration.shared.sendWithStacktrace(message: "Got status \(status) while attempting to close_v2.", tag: "SwiftData", severity: .error)
+                return createErr("During: closing database with flags", status: Int(status))
```

Comment:
Maybe we should clarify this error and make clear that this was attempted with `close_v2`?